datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -138,12 +138,21 @@ impl FileScanConfig {
             column_statistics: table_cols_stats,
         };
 
-        let table_schema = Arc::new(
+        let projected_schema = Arc::new(
             Schema::new(table_fields).with_metadata(self.file_schema.metadata().clone()),
         );
+
+        let full_table_schema = {
```

Comment:
I wonder if we can avoid creating the entire schema here and instead reuse it from elsewhere -- the information should already be present to plan queries I think. As we try to make planning more efficient avoiding re-creating the same schema over and over again I think will be importatn

Reply:
I managed to get rid of creating this schema, fortunately -- we can use projection (the `Vec<usize>`) to recover the positions of the statistics in the original schema.